frontend/src/views/HomeView.vue

```diff
@@ -1 +1,60 @@
-<template><h1>Test</h1></template>
+<script setup lang="ts">
+import { ElButton, ElInput, ElMessage, ElDialog} from 'element-plus'
+import { ref } from 'vue'
+import { patch } from '@/services/http';
+
+
+const dialogFormVisible = ref(false)
+const editRole = ref({name:''})
+
+
+const onClickEditRole = () => {
+  dialogFormVisible.value = true
+}
+
+const onClickCancel = () => {
+  dialogFormVisible.value = false
+}
+
+const saveForm = async (roleId: number) => {
+  try {
+    const response: { message?: string } = await patch(`/api/roles/${roleId}`, editRole.value);
```

Comment:
You have to create a function inside actions file, in state management. This function will responsible for

1. calling the backend (so, you'll not do this here)
2. Edit the role in the state

Here you should just import this action, call it, and nothing more. The action should be responsible for this

Check in Demo how an action is called, to fetch the roles, and base your work like that